Drop-In/src/main/java/com/braintreepayments/api/DropInClient.java

```diff
@@ -0,0 +1,428 @@
+     * payment method, you may not need to show Drop-In.
+     * <p>
+     * Note: a client token must be used and will only return a payment method if it contains a
+     * customer id.
+     *
+     * @param activity the current {@link AppCompatActivity}
+     * @param callback callback for handling result
+     */
+    public void fetchMostRecentPaymentMethod(FragmentActivity activity, final FetchMostRecentPaymentMethodCallback callback) {
+        // TODO: consider sending back empty result if tokenization key auth
+        boolean isClientToken = braintreeClient.getAuthorization() instanceof ClientToken;
+        if (!isClientToken) {
+            InvalidArgumentException error = new InvalidArgumentException("DropInClient#fetchMostRecentPaymentMethods() must " +
+                    "be called with a client token");
+            callback.onResult(null, error);
+            return;
+        }
+
+        final DropInPaymentMethodType lastUsedPaymentMethodType = DropInPaymentMethodType.forType(BraintreeSharedPreferences.getSharedPreferences(activity)
+                .getString(LAST_USED_PAYMENT_METHOD_TYPE, null));
+
+        if (lastUsedPaymentMethodType == DropInPaymentMethodType.GOOGLE_PAYMENT) {
+            googlePayClient.isReadyToPay(activity, new GooglePayIsReadyToPayCallback() {
+                @Override
+                public void onResult(Boolean isReadyToPay, Exception error) {
+                    if (isReadyToPay) {
+                        DropInResult result = new DropInResult();
+                        result.setPaymentMethodType(lastUsedPaymentMethodType);
+                        callback.onResult(result, null);
+                    } else {
+                        getPaymentMethodNonces(callback);
+                    }
+                }
+            });
+        } else {
+            getPaymentMethodNonces(callback);
+        }
+    }
+
+    private void getPaymentMethodNonces(final FetchMostRecentPaymentMethodCallback callback) {
+        paymentMethodClient.getPaymentMethodNonces(new GetPaymentMethodNoncesCallback() {
+            @Override
+            public void onResult(@Nullable List<PaymentMethodNonce> paymentMethodNonceList, @Nullable Exception error) {
+                if (paymentMethodNonceList != null) {
+                    DropInResult result = new DropInResult();
+                    if (paymentMethodNonceList.size() > 0) {
+                        PaymentMethodNonce paymentMethod = paymentMethodNonceList.get(0);
+                        result.paymentMethodNonce(paymentMethod);
+                    }
+                    callback.onResult(result, null);
+                } else if (error != null) {
+                    callback.onResult(null, error);
+                }
+            }
+        });
+    }
+
+    void getVaultedPaymentMethods(final FragmentActivity activity, boolean refetch, final GetPaymentMethodNoncesCallback callback) {
```

Comment:
Will all docstrings be a separate PR?

Reply:
Yes we want to wait to finalize the methods/API before we do docstrings. We added a TODO in `DropInClient` to add docstrings.